components/User/UserProfile/Integrations.js

```diff
@@ -0,0 +1,390 @@
+                }
+            });
+        // console.log('Integrations render()', this.props.UserProfileStore.user.providers);
+
+        let facebook_icon_classes = classNames({
+            'big': true,
+            'facebook': true,
+            'square': true,
+            'disabled': !facebook,
+            'icon': true,
+        });
+        let facebook_enable_classes = classNames({
+            'ui': true,
+            'positive': !facebook,
+            'disabled': facebook,
+            'button': true,
+        });
+        let facebook_disable_classes = classNames({
+            'ui': true,
+            'negative': facebook,
+            'disabled': !facebook,
+            'button': true,
+        });
+        let google_icon_classes = classNames({
+            'big': true,
+            'google': true,
+            'plus': true,
+            'disabled': !google,
+            'icon': true,
+        });
+        let google_enable_classes = classNames({
+            'ui': true,
+            'positive': !google,
+            'disabled': google,
+            'button': true,
+        });
+        let google_disable_classes = classNames({
+            'ui': true,
+            'negative': google,
+            'disabled': !google,
+            'button': true,
+        });
+        let github_icon_classes = classNames({
+            'big': true,
+            'github': true,
+            'disabled': !github,
+            'icon': true,
+        });
+        let github_enable_classes = classNames({
+            'ui': true,
+            'positive': !github,
+            'disabled': github,
+            'button': true,
+        });
+        let github_disable_classes = classNames({
+            'ui': true,
+            'negative': github,
+            'disabled': !github,
+            'button': true,
+        });
+
+        let text_providerEnabled = 'This provider is enabled and could be used.';
+        let text_providerDisabled = 'If you want to sign in later with this provider, you have to click the Enable button.';
+
+        return (
+          <div>
+              <div className="ui segments">
+
+                  <div className="ui secondary segment">
+                      <h3>Hint</h3>
+                  </div>
+                  <div className="ui segment">
+                    <p>
+                      SlideWiki provides the possibility to sign in with multiple providers.
+                      In order to be able to use a specific provider you have to active it.
+                      Activating a provider will open a new tab for a sign in there.
+                      Please sign in there and don&apos;t close the tab.
+                    </p>
+                  </div>
+
+              </div>
+              <div className="ui segments">
+
+                  <div className="ui secondary segment">
+                      <h3>Login Provider</h3>
+                  </div>
+                  <div className="ui segment">
```

Comment:
The following paragraphs have the exact same structure. Use a loop to cycle through the list of all providers (currently three) instead writing it three times.